src/NUnitFramework/framework/Constraints/CollectionTally.cs

```diff
@@ -62,35 +83,26 @@ private bool ItemsEqual(object expected, object actual)
             return comparer.AreEqual(expected, actual, ref tolerance);
         }
 
-        /// <summary>
-        /// Try to remove an object from the tally
-        /// </summary>
-        /// <param name="o">The object to remove</param>
-        /// <returns>True if successful, false if the object was not found</returns>
-        public bool TryRemove(object o)
+        /// <summary>Try to remove an object from the tally.</summary>
+        /// <param name="o">The object to remove.</param>
+        public void TryRemove(object o)
         {
-            for (int index = 0; index < list.Count; index++)
-                if (ItemsEqual(list[index], o))
+            for (int index = 0; index < _missingItems.Count; index++)
+                if (ItemsEqual(_missingItems[index], o))
                 {
-                    list.RemoveAt(index);
-                    return true;
+                    _missingItems.RemoveAt(index);
+                    return;
                 }
 
-            return false;
+            _extraItems.Add(o);
         }
 
-        /// <summary>
-        /// Try to remove a set of objects from the tally
-        /// </summary>
-        /// <param name="c">The objects to remove</param>
-        /// <returns>True if successful, false if any object was not found</returns>
-        public bool TryRemove(IEnumerable c)
+        /// <summary>Try to remove a set of objects from the tally.</summary>
+        /// <param name="c">The objects to remove.</param>
+        public void TryRemove(IEnumerable c)
```

Comment:
Note for future PR: we gotta be careful depending where this class is used because `TryRemove(stringExpression)` will match `TryRemove(IEnumerable)`, not `TryRemove(object)`.